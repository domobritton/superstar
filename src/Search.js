import React from 'react';

import TextField from '@material-ui/core/TextField';
import styled from 'styled-components';

export const Search = ({ text, updateSearch, value }) => (
  <Form noValidate autoComplete="off">
    {value !== 0 && (
      <TextField
        label={text}
        placeholder="What will you star today?"
        fullWidth
        onChange={updateSearch}
        InputLabelProps={{
          shrink: true,
        }}
      />
    )}
  </Form>
);

const Form = styled.form`
  display: flex;
  padding: 40px 5%;
  height: 60px;
`;
