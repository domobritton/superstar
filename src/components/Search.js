import React from 'react';
import PropTypes from 'prop-types';

import TextField from '@material-ui/core/TextField';
import styled from 'styled-components';

export const Search = ({ text, updateSearch, value }) => (
  <Form
    noValidate
    autoComplete="off"
    onSubmit={e => {
      e.preventDefault();
    }}
  >
    {value !== 0 && (
      <TextField
        label={text}
        placeholder="What will you star today?"
        fullWidth
        onChange={updateSearch}
        InputLabelProps={{
          shrink: true,
        }}
      />
    )}
  </Form>
);

Search.propTypes = {
  text: PropTypes.string,
  updateSearch: PropTypes.func,
  value: PropTypes.number,
};

const Form = styled.form`
  display: flex;
  padding: 40px 5%;
  height: 160px;
`;
