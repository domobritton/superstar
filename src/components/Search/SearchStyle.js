import styled from 'styled-components';

export const Form = styled.form`
  display: flex;
  padding: 40px 5%;
  height: 160px;
`;
