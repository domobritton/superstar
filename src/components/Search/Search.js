import React from 'react';
import PropTypes from 'prop-types';

import TextField from '@material-ui/core/TextField';
import { Form } from './SearchStyle';

export const Search = ({ text, updateSearch, value }) => {
  const placeholder =
    value === 0 ? 'Enter username' : 'What will you star today?';
  return (
    <Form
      noValidate
      autoComplete="off"
      onSubmit={e => {
        e.preventDefault();
      }}
    >
      <TextField
        label={text}
        placeholder={placeholder}
        fullWidth
        onChange={updateSearch}
        InputLabelProps={{
          shrink: true,
        }}
      />
    </Form>
  );
};

Search.propTypes = {
  text: PropTypes.string,
  updateSearch: PropTypes.func,
  value: PropTypes.number,
};
