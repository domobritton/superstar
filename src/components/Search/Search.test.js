import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Search } from './Search';

afterEach(cleanup);

test('<Search />', () => {
  const { container } = render(<Search />);
  expect(container.firstChild).toMatchSnapshot();
});
