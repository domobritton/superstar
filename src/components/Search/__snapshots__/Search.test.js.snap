// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> 1`] = `
<form
  autocomplete="off"
  class="sc-bdVaJa btorsK"
  novalidate=""
>
  <div
    class="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  >
    <div
      class="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-fullWidth-27 MuiInput-fullWidth-12 MuiInputBase-formControl-19 MuiInput-formControl-6"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input-28 MuiInput-input-13"
        placeholder="What will you star today?"
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
